diffkemp/simpll/Utils.h

```diff
@@ -118,4 +118,15 @@ std::string getIdentifierForValue(const Value *Val,
         const std::map<std::pair<StructType *, uint64_t>, StringRef>
         &StructFieldNames, const Function *Parent = nullptr);
 
+/// Takes two vectors and creates a vector of pairs from them.
+template<typename T> std::vector<std::pair<T, T>> pairZip(std::vector<T> A,
```

Comment:
LLVM provides llvm::zip located in ADT/STLExtras.h. Could we use it instead of writing our own zip?
Here's the PR that introduced it, there are tests with usage examples as well:
https://reviews.llvm.org/D23252